lib/rules/no-observers.js

```diff
@@ -30,40 +31,67 @@ module.exports = {
       context.report(node, ERROR_MESSAGE);
     };
 
+    let importedEmberName = undefined;
+    let importedObserverName = undefined;
+    let importedObservesName = undefined;
+    let importedAddObserverName = undefined;
+
     return {
       CallExpression(node) {
-        if (ember.isModule(node, 'observer')) {
-          report(node);
-        }
-
         const { callee } = node;
-
         if (
-          (types.isThisExpression(callee.object) || types.isMemberExpression(callee)) &&
+          // observer();
+          types.isIdentifier(callee) &&
+          callee.name === importedObserverName
+        ) {
+          report(node);
+        } else if (
+          // Ember.observer();
+          types.isMemberExpression(callee) &&
+          types.isIdentifier(callee.object) &&
```

Comment:
Probably not in scope for this, but I think it would be nice if you could do `types.isIdentifier(callee.object, importedEmberName)` to check if it's a particular identifier.

Reply:
Yeah, we had some helpers like that in eslint-plugin-square. It would be cool if someone created an "ast utils" library for convenient helpers like that. However, I actually would like to get rid of the current type check functions since they are pretty pointless today in that they just making a single string comparison.